hydra-node/test/Hydra/Network/HeartbeatSpec.hs

```diff
@@ -5,53 +5,108 @@ import Hydra.Prelude
 import Control.Monad.Class.MonadSTM (modifyTVar', newTVarIO)
 import Control.Monad.IOSim (runSimOrThrow)
 import Hydra.HeadLogic (HydraMessage (..))
-import Hydra.Ledger.Simple (SimpleTx)
-import Hydra.Network (Host (..), Network (..))
-import Hydra.Network.Heartbeat (withHeartbeat)
+import Hydra.Network (
+  Host (..),
+  Network (..),
+ )
+import Hydra.Network.Heartbeat (Heartbeat (..), withHeartbeat)
 import Test.Hspec (Spec, describe, it, shouldBe)
 
 spec :: Spec
 spec = describe "Heartbeat" $ do
-  let dummyNetwork msgqueue _cb action =
+  let captureOutgoing msgqueue _cb action =
         action $ Network{broadcast = \msg -> atomically $ modifyTVar' msgqueue (msg :)}
 
-  it "sends a heartbeat message with own party id every 500 ms" $ do
+      captureIncoming receivedMessages msg =
+        atomically $ modifyTVar' receivedMessages (msg :)
+
+  it "sends a heartbeat message with own party id after 500 ms" $ do
     let sentHeartbeats = runSimOrThrow $ do
-          sentMessages <- newTVarIO ([] :: [HydraMessage SimpleTx])
+          sentMessages <- newTVarIO ([] :: [Heartbeat (HydraMessage Integer)])
 
-          withHeartbeat testHost (dummyNetwork sentMessages) noop $ \_ -> do
+          withHeartbeat 1 (captureOutgoing sentMessages) noop $ \_ ->
             threadDelay 1.1
 
           readTVarIO sentMessages
 
-    sentHeartbeats `shouldBe` replicate 2 (Ping testHost)
+    sentHeartbeats `shouldBe` [Ping 1]
```

Comment:
Having the assertions in the `runSimOrThrow` would be quite a bit easier to read, i.e
```haskell
    runSimOrThrow $ do
          sentMessages <- newTVarIO ([] :: [Heartbeat (HydraMessage Integer)])

          withHeartbeat 1 (captureOutgoing sentMessages) noop $ \_ ->
            threadDelay 1.1

          sentHeartbeats <- readTVarIO sentMessages
          sentHeartbeats `shouldBe` [Ping 1]
```

Where the `newTVarIO` and `readTVarIO` could be also moved into (typed) boilerplate functions with drafted usage:

```haskell
    runSimOrThrow $ do
          (capture, getSentHeartbeats) <- captureOutgoing

          withHeartbeat 1 capture noop $ \_ ->
            threadDelay 1.1

          getSentHeartbeats `shouldReturn` [Ping 1]
```